File: src/main/java/lesson/java/basics/Exceptions.java
package lesson.java.basics;

public class Exceptions {
    public static void main(String args[]){
        try{
            throw new IndexOutOfBoundsException("Sample Exception");
            //throw new NullPointerException("Sample Exception");
            //throw new Exception("Sample Exception");
        }catch(IndexOutOfBoundsException e){
            System.out.println(e.toString());
        }catch(NullPointerException e){
            System.out.println(e.toString());
        }catch(Exception e){
            System.out.println(e.toString());
        }finally{
            System.out.println("done");
        }
    }
}

File: src/main/scala/lesson/scala/basics/Exceptions.scala
package lesson.scala.basics

object Exceptions {
  def main(args: Array[String]) = {
    try{
      //throw new IndexOutOfBoundsException("Sample Exception")
      //throw new NullPointerException("Sample Exception")
      //throw new Exception("Sample Exception")
    }catch{
      case e: IndexOutOfBoundsException => println(e.toString)
      case e: NullPointerException => println(e.toString)
      case e: Exception => println(e.toString)
    }finally{
      println("done");
    }
  }
}
